### ISSUE TITLE
Referer related issue

### ISSUE THREAD

**Issue by mazesy** [cowlicks/privacypossum#25]
Privacy Possum 2018.5.7 / Firefox 59.0.3 64-BIT / Windows 7 64-BIT

Among Privacy Possum' features, _blocks `refer` headers that reveal your browsing location_

I have a[ test site](http://www.ancreport.com/podcast/ian-williams-israel-lobby-u-n/) for checking referers (used with dedicated referer extensions). This site calls a Vimeo video (embedded) and Vimeo requires the calling site's referer. PP had to be disabled for that site. The `referer` was not mentioned in PP's popup so I had to disable PP globally for that site

Referer management requires whitelists because if a global strict Referer policy makes it on most sites, some others won't display a component without the referer such as above mentioned. I state this because it seems to be a pity if we have to disable all PP for a site when only the rerefer is concerned (and did not appear in above mentioned page's PP popup).

**Comment by cowlicks** [comment]
We should definitely have a per-header type deactivation eventually.

Instead of a whitelist I'd prefer to have a heuristic to catch or allow cases like this. Most content that requires `referer` is embed-able stuff, like `iframes`.

I did a quick test and found that vimeo allows video's to be embedded when the referer is allowed for *just* the iframe request. However allowing `referer` on all iframes is not an insignificant privacy compromise.

I have a few other things I'd like to explore here. Like changing the referer from the host site to the requested site. So for your test case we could try changing the refere to vimeo.com/whatever-video

**Comment by cowlicks** [comment]
Here is a diff for that experiment

```diff
diff --git a/src/js/reasons/headers.js b/src/js/reasons/headers.js
index 12765a5..1b1859c 100644
--- a/src/js/reasons/headers.js
+++ b/src/js/reasons/headers.js
@@ -8,7 +8,14 @@ const {Action} = require('../schemes'),
   {etagHeader} = require('./etag'),
   {HEADER_DEACTIVATE_ON_HOST, header_methods, NO_ACTION, TAB_DEACTIVATE_HEADERS} = require('../constants');
 
-const alwaysTrue = () => true;
+const alwaysTrue = () => true,
+    refererHandler = (details, header) => {
+      if (details.type === 'sub_frame') {
+        console.log(details);
+        return false;
+      }
+      return true;
+    };
 
 class HeaderHandler {
   constructor(store) {
@@ -16,7 +23,7 @@ class HeaderHandler {
     this.badHeaders = new Map([
       ['cookie', alwaysTrue],
       ['set-cookie', alwaysTrue],
-      ['referer', alwaysTrue],
+      ['referer', refererHandler],
       ['etag', etagHeader.bind(undefined, {store, cache})],
       ['if-none-match', alwaysTrue]
     ]);
```

**Comment by cowlicks** [comment]
Closed by #66 ! I still no whitelist. If the site returns a 4XX when the referer is stripped, we try again without stripping the referer header. This works for vimeo.
